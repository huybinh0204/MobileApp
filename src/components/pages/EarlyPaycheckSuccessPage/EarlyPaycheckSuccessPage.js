import { useNavigation } from '@react-navigation/native';
import React, { useRef } from 'react';
import ENV from 'react-native-config';
import BottomSheet from 'react-native-raw-bottom-sheet';
import { useDispatch, useSelector } from 'react-redux';
import { SecondaryScreenLayout } from '_components/organisms';
import { ICONS, NAVIGATION } from '_constants';
import strings from '_localization';
import { AuthenticationActions, AuthenticationSelectors } from '_store/authentication';
import { normalize } from '_utilities/screen';
import {
  BottomSheetButton,
  BottomTitle,
  Button,
  Card,
  CardBody,
  CardContainer,
  CardTitle,
  Container,
  ContentTitle,
  Description,
  DescriptionLink,
  SheetContainer,
  StatusIcon,
  TextLink,
} from './EarlyPaycheckSuccessPage.styles';

const EarlyPaycheckSuccessPage = () => {
  const { navigate } = useNavigation();
  const bottomSheetRef = useRef(null);
  const dispatch = useDispatch();
  const isSignIn = useSelector(AuthenticationSelectors.getIsSignIn);

  const onPress = () => {
    if (isSignIn) {
      navigate(NAVIGATION.shared.main);
    } else {
      dispatch(AuthenticationActions.setIsSignIn(true));
    }
  };

  const handleLink = () => {
    bottomSheetRef?.current.open();
  };

  return (
    <SecondaryScreenLayout>
      <Container>
        <StatusIcon icon={ICONS.checkmarkLines} width={normalize(60)} height={normalize(60)} />
        <ContentTitle>{strings.earlyPaycheck.success.title}</ContentTitle>
        <Card>
          <StatusIcon
            icon={ICONS.overdraftProtection}
            width={normalize(60)}
            height={normalize(60)}
          />
          <CardContainer>
            <CardTitle>{strings.overdraft.sell.title}</CardTitle>
            <CardBody>
              {strings.formatString(strings.earlyPaycheck.success.description, {
                learnMoreLink: (
                  <TextLink onPress={handleLink}>
                    {strings.earlyPaycheck.success.learnMore}
                  </TextLink>
                ),
              })}
            </CardBody>
          </CardContainer>
        </Card>
        <Button onPress={onPress} testID="earlyPaycheckSuccessButton">
          {strings.earlyPaycheck.success.buttonText}
        </Button>
      </Container>
      <BottomSheet closeOnPressMask height={normalize(480)} ref={bottomSheetRef}>
        <SheetContainer>
          <BottomTitle>{strings.overdraftProtection.successBottomSheet.title}</BottomTitle>
          <Description>{strings.overdraftProtection.successBottomSheet.description}</Description>
          <Description>
            {strings.formatString(strings.overdraftProtection.successBottomSheet.disclaimer, {
              disclaimerLink: (
                <DescriptionLink href={ENV.OVERDRAFT_LEGAL_AGREEMENT_URL}>
                  {strings.overdraftProtection.successBottomSheet.disclaimerLinkText}
                </DescriptionLink>
              ),
            })}
          </Description>
          <BottomSheetButton
            onPress={() => bottomSheetRef.current?.close()}
            testID="overdraftProtectionGotItButton"
          >
            {strings.overdraftProtection.successBottomSheet.buttonText}
          </BottomSheetButton>
        </SheetContainer>
      </BottomSheet>
    </SecondaryScreenLayout>
  );
};

export default EarlyPaycheckSuccessPage;
