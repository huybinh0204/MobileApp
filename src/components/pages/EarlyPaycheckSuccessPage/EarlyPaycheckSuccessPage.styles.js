import styled from '@emotion/native';
import { BaseText, IconSvg, Link, MainButton } from '_components/atoms';
import { normalize } from '_utilities/screen';

export const Container = styled.View(({ theme }) => ({
  flex: 1,
  padding: theme.spacing.m,
}));

export const StatusIcon = styled(IconSvg)({
  alignSelf: 'center',
});

export const ContentTitle = styled(BaseText)(({ theme }) => ({
  ...theme.typography.heading4,
  color: theme.colors.beta900,
  marginTop: theme.spacing.l,
  marginBottom: theme.spacing.s,
  textAlign: 'center',
}));

export const Content = styled(BaseText)(({ theme }) => ({
  ...theme.typography.subContentRegular,
  color: theme.colors.beta500,
  textAlign: 'center',
}));

export const Button = styled(MainButton)({
  marginTop: 'auto',
});

export const CardTitle = styled(BaseText)(({ theme }) => ({
  ...theme.typography.subContentSemiBold,
  color: theme.colors.beta900,
  marginBottom: theme.spacing.xss,
}));

export const CardBody = styled(BaseText)(({ theme }) => ({
  ...theme.typography.subContentRegular,
  color: theme.colors.beta900,
}));

export const Card = styled.View(({ theme }) => ({
  backgroundColor: theme.colors.white,
  borderRadius: 10,
  borderColor: theme.colors.beta100,
  borderWidth: 1,
  marginVertical: theme.spacing.xs,
  padding: theme.spacing.s,
  flexDirection: 'row',
  height: normalize(140),
}));

export const TextLink = styled(BaseText)(({ theme }) => ({
  ...theme.typography.subContentRegular,
  color: theme.colors.alpha500,
}));

export const CardContainer = styled.View(({ theme }) => ({
  paddingLeft: theme.spacing.s,
  flex: 1,
}));

export const SheetContainer = styled.View(({ theme }) => ({
  flex: 1,
  padding: theme.spacing.m,
}));

export const BottomTitle = styled(BaseText)(({ theme }) => ({
  ...theme.typography.heading5,
  color: theme.colors.beta900,
  marginBottom: theme.spacing.xs,
}));

export const Description = styled(BaseText)(({ theme }) => ({
  ...theme.typography.mainContentRegular,
  color: theme.colors.beta400,
  marginBottom: theme.spacing.s,
}));

export const DescriptionLink = styled(Link)(({ theme }) => ({
  ...theme.typography.mainContentRegular,
  color: theme.colors.alpha500,
}));

export const BottomSheetButton = styled(MainButton)(({ theme }) => ({
  marginTop: theme.spacing.m,
}));
