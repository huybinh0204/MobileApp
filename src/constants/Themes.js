import { DarkTheme as Dark, DefaultTheme as Default } from '@react-navigation/native';
import colors from './Colors';
import fontSize from './FontSizes';
import spacing from './Spacing';
import typography from './Typography';

const ColorScheme = {
  LIGHT: 'light',
  DARK: 'dark',
};

const LightTheme = {
  ...Default,
  colors: {
    ...Default.colors,
    ...colors,
    textColor: colors.beta900,
  },
  spacing,
  fontSize,
  typography,
};

const DarkTheme = {
  ...Dark,
  colors: {
    ...Dark.colors,
    ...colors,
    textColor: colors.white,
  },
  spacing,
  fontSize,
  typography,
};

export { ColorScheme, LightTheme, DarkTheme };
