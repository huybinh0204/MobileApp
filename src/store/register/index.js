export * from './register_reducer';
export * from './register_sagas';
