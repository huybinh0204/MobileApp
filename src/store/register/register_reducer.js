import { createActions, createReducer } from 'reduxsauce';
import Immutable from 'seamless-immutable';

const { Types, Creators } = createActions(
  {
    trackEvent: ['event', 'eventType', 'properties'],
    trackEventSuccess: ['message'],
    trackEventFails: ['message'],
  },
  {
    prefix: '@@BE-REGISTER/',
  }
);

export const RegisterActions = Creators;
export const RegisterTypes = Types;

// INITIAL STATE
export const INITIAL_STATE = Immutable({});

// ACTION HANDLERS
const trackEventSuccess = (state, { message }) => {
  console.info('Track Event SUCCESS:', message);
  return state;
};

const trackEventFails = (state, { message }) => {
  console.warn('Track Event FAIL:', message);
  return state;
};

// REDUCER
export const RegisterReducer = createReducer(INITIAL_STATE, {
  [Types.TRACK_EVENT_SUCCESS]: trackEventSuccess,
  [Types.TRACK_EVENT_FAILS]: trackEventFails,
});
