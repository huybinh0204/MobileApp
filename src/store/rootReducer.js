import { combineReducers } from 'redux';
import actionWatchReducer from 'redux-action-watch/lib/reducer';
import { resettableReducer } from 'reduxsauce';
import { AccountReducer } from './account';
import { AuthenticationReducer } from './authentication';
import { CustomerReducer } from './customer';
import { FundDirectDepositStoreReducer } from './pageStore/fundDirectDepositStore';
import { RegisterReducer } from './register';
import { TransferReducer } from './transfer';

const resettable = resettableReducer('RESET');

export default combineReducers({
  account: resettable(AccountReducer),
  auth: resettable(AuthenticationReducer),
  customer: resettable(CustomerReducer),
  fundDirectDepositStoreReducer: resettable(FundDirectDepositStoreReducer),
  register: RegisterReducer,
  transfer: resettable(TransferReducer),
  watcher: actionWatchReducer, // Must be last
});
