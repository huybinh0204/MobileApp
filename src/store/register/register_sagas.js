import { call, put, select } from 'redux-saga/effects';
import { EVENT_TYPES } from '_constants';
import { CustomerSelectors } from '_store/customer';
import { RegisterActions } from './register_reducer';

export function* trackEvent(api, action) {
  const { event, properties = {}, eventType = EVENT_TYPES.SCREEN } = action;
  const customerExternalId = yield select(CustomerSelectors.getExternalId);
  const userId = customerExternalId ?? 'unknown';

  if (!EVENT_TYPES[eventType]) {
    console.warn(`Invalid argument: ${eventType} is not a valid eventType`);
  }

  try {
    yield call(api.sendEvent, event, properties, eventType, userId);
    yield put(RegisterActions.trackEventSuccess(event));
  } catch (e) {
    yield put(RegisterActions.trackEventFails(`${event} with ${e}`));
  }
}
