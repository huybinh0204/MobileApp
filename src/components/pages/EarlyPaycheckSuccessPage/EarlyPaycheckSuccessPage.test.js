import { ThemeProvider } from '@emotion/react';
import { useNavigation } from '@react-navigation/native';
import { fireEvent, render } from '@testing-library/react-native';
import React from 'react';
import * as ReactRedux from 'react-redux';
import { createStore } from 'redux';
import { NAVIGATION } from '_constants';
import { LightTheme } from '_constants/Themes';
import { RegisterActions } from '_store/register';
import rootReducer from '_store/rootReducer';
import { navigationMock } from '_test/mocks/navigation';
import EarlyPaycheckSuccessPage from './EarlyPaycheckSuccessPage';

jest.spyOn(RegisterActions, 'trackEvent');
jest.mock('@react-navigation/native');

describe('EarlyPaycheckSuccessPage', () => {
  let mockStore;
  let mockNavigation;
  const dispatch = jest.fn();
  jest.spyOn(ReactRedux, 'useDispatch').mockReturnValue(dispatch);

  const RenderDirectDepositFormStatusPage = (props) => (
    <ReactRedux.Provider store={mockStore}>
      <ThemeProvider theme={LightTheme}>
        <EarlyPaycheckSuccessPage {...props} />
      </ThemeProvider>
    </ReactRedux.Provider>
  );
  beforeEach(() => {
    mockNavigation = navigationMock(jest.fn);
    useNavigation.mockImplementation(() => mockNavigation);
    mockStore = createStore(rootReducer, {});
  });
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match snapshot', () => {
    const { toJSON } = render(<RenderDirectDepositFormStatusPage />);
    expect(toJSON()).toMatchSnapshot();
  });

  describe('Navigate to hometab when the button is pressed', () => {
    it('should dispatch isSignIn action to navigate to hometab if isSignIn is false', async () => {
      ReactRedux.useSelector = jest.fn().mockReturnValue(false);
      ReactRedux.useDispatch = jest.fn().mockReturnValue(dispatch);
      const { findByTestId } = render(<RenderDirectDepositFormStatusPage />);
      const directDepositFormStatusButton = await findByTestId('earlyPaycheckSuccessButton');
      fireEvent.press(directDepositFormStatusButton);
      expect(dispatch).toHaveBeenCalledWith({
        isSignIn: true,
        type: '@@BE-AUTHENTICATION/SET_IS_SIGN_IN',
      });
    });

    it('should use navigate to navigate to hometab if isSignIn is true', async () => {
      ReactRedux.useSelector = jest.fn().mockReturnValue(true);
      const { findByTestId } = render(<RenderDirectDepositFormStatusPage />);
      const directDepositFormStatusButton = await findByTestId('earlyPaycheckSuccessButton');
      fireEvent.press(directDepositFormStatusButton);
      expect(mockNavigation.navigate).toBeCalledWith(NAVIGATION.shared.main);
    });
  });
});
