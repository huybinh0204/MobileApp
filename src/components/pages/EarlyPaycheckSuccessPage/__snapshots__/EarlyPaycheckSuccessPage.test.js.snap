// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EarlyPaycheckSuccessPage should match snapshot 1`] = `
<RCTSafeAreaView
  backgroundColor={null}
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "#FAFAFA",
        "flex": 1,
      },
      undefined,
    ]
  }
  testID="SecondaryScreenLayout"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "height": 108.5,
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "left": 36,
          "opacity": 1,
          "position": "absolute",
          "zIndex": 10,
        }
      }
      testID="BackButton"
    >
      <SvgMock
        height={45.5}
        width={45.5}
      />
    </View>
  </View>
  <View
    onLayout={[Function]}
    style={
      Array [
        Array [
          Object {
            "flex": 1,
          },
          undefined,
        ],
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "padding": 58,
          },
          undefined,
        ]
      }
    >
      <SvgMock
        height={108.5}
        style={
          Array [
            Object {
              "alignSelf": "center",
            },
            undefined,
          ]
        }
        width={108.5}
      />
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#1F1D25",
              "fontFamily": "Poppins-SemiBold",
              "fontSize": 43.5,
              "lineHeight": 60,
              "marginBottom": 29,
              "marginTop": 87,
              "textAlign": "center",
            },
            undefined,
          ]
        }
      >
        Congratulations!
Direct deposit is set up
      </Text>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#D4D7DB",
              "borderRadius": 10,
              "borderWidth": 1,
              "flexDirection": "row",
              "height": 253.5,
              "marginVertical": 14.5,
              "padding": 29,
            },
            undefined,
          ]
        }
      >
        <SvgMock
          height={108.5}
          style={
            Array [
              Object {
                "alignSelf": "center",
              },
              undefined,
            ]
          }
          width={108.5}
        />
        <View
          style={
            Array [
              Object {
                "flex": 1,
                "paddingLeft": 29,
              },
              undefined,
            ]
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#1F1D25",
                  "fontFamily": "Poppins-SemiBold",
                  "fontSize": 25.5,
                  "lineHeight": 35.5,
                  "marginBottom": 3.5,
                },
                undefined,
              ]
            }
          >
            Kinly Overdraft
          </Text>
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#1F1D25",
                  "fontFamily": "Poppins",
                  "fontSize": 25.5,
                  "lineHeight": 35.5,
                },
                undefined,
              ]
            }
          >
            Continue receiving direct deposits of at least $1,000 over 35 calendar days. We’ll let you know when it’s ready! [object Object]
          </Text>
        </View>
      </View>
      <View
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "stretch",
            "backgroundColor": "#3D3397",
            "borderRadius": 10,
            "marginTop": "auto",
            "opacity": 1,
            "paddingHorizontal": 45.5,
            "paddingVertical": 31,
          }
        }
        testID="earlyPaycheckSuccessButton"
      >
        <Text
          allowFontScaling={false}
          disabled={false}
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontFamily": "Poppins-SemiBold",
                "fontSize": 25.5,
                "lineHeight": 35.5,
                "textAlign": "center",
              },
              undefined,
            ]
          }
        >
          Got it
        </Text>
      </View>
    </View>
    <Modal
      animationType="none"
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      supportedOrientations={
        Array [
          "portrait",
          "portrait-upside-down",
          "landscape",
          "landscape-left",
          "landscape-right",
        ]
      }
      transparent={true}
      visible={false}
    />
  </View>
</RCTSafeAreaView>
`;
